engine/inc/com/centreon/engine/configuration/applier/pb_difference.hh

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright 2023 Centreon
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ */
+
+#ifndef CCE_CONFIGURATION_APPLIER_PB_DIFFERENCE_HH
+#define CCE_CONFIGURATION_APPLIER_PB_DIFFERENCE_HH
+
+#include <absl/container/flat_hash_map.h>
+#include <absl/container/flat_hash_set.h>
+#include <google/protobuf/util/message_differencer.h>
+
+#include <iterator>
+
+namespace com::centreon::engine {
+
+using MessageDifferencer = ::google::protobuf::util::MessageDifferencer;
+
+namespace configuration {
+namespace applier {
+template <typename T,
+          typename Key,
+          typename Container = ::google::protobuf::RepeatedPtrField<const T>>
+class pb_difference {
+  // What are the new objects
+  std::vector<T> _added;
```

Comment:
Is it mandatory to copy all new element? no means to use a const T *?

Reply:
I will take a look. Maybe this is a bad influence of the legacy code...